doc/tutorial/search/solution5.cpp

```diff
@@ -0,0 +1,80 @@
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/algorithm/search.hpp>
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+
+using namespace seqan3;
+using namespace seqan3::search_cfg;
+using namespace seqan3::align_cfg;
+
+void run_text_single()
+{
+    dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+    dna4_vector query{"GCT"_dna4};
+    fm_index index{text};
+
+    using iter_t = decltype(std::begin(text));
+
+    debug_stream << "Searching all best hits allowing for 1 error in a single text\n";
+
+    configuration const search_cfg = search_cfg::max_error{total{1}} | search_cfg::mode{all_best};
+    configuration const align_cfg = edit /*| aligned_ends{all_ends_free}*/ | result{with_trace};
+    auto results = search(index, query, search_cfg);
+    debug_stream << "There are " << results.size() << " hits.\n";
+
+    for (auto && pos : results)
+    {
+        size_t start = pos ? pos - 1 : 0;
+        size_t end   = pos + query.size() + 1;
+        auto text_view = view::subrange<iter_t, iter_t>{std::begin(text) + start, std::begin(text) + end};
+
+        auto align_sequences = std::make_pair(text_view, query);
+
+        for (auto && alignment : align_pairwise(align_sequences, align_cfg))
+        {
+            auto aligned_sequence = alignment.get_alignment();
+            debug_stream << "score:\t\t " << alignment.get_score() << '\n';
+            debug_stream << "gapped_database: " << get<0>(aligned_sequence) << '\n';
+            debug_stream << "gapped_query:\t "  << get<1>(aligned_sequence) << '\n';
+        }
+    }
+}
+
+void run_text_collection()
+{
+    std::vector<dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
+                                  "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
+                                  "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+    dna4_vector query{"GCT"_dna4};
+    fm_index index{text};
+
+    using iter_t = decltype(std::begin(*std::begin(text)));
+
+    debug_stream << "Searching all best hits allowing for 1 error in a text collection\n";
+
+    configuration const search_config = search_cfg::max_error{total{1}} | search_cfg::mode{all_best};
```

Comment:
here you are using the namespace labels explicitly, but not in other places... I think it would be better to have them everywhere, but at least it should be consistent.

Reply:
So, there are a few possibilities:

1) always use explicitly
```cpp
configuration const cfg_all = search_cfg::max_error{search_cfg::total{1}} | search_cfg::mode{search_cfg::all};
```

2) use implicitly
```cpp
using namespace search_cfg;
configuration const cfg_all = max_error{total{1}} | mode{all};
```

3) mixed
```cpp
using namespace search_cfg;
configuration const cfg_all = search_cfg::max_error{total{1}} |search_cfg::mode{all};
```

When also using the `align_cfg` I need to use at least 3) to make it not ambiguous (`max_error` is in both).

Reply:
I would always write explicitly for now.